.github/workflows/tests-clustermesh-upgrade.yaml

```diff
@@ -449,6 +509,23 @@ jobs:
           cilium --context ${{ env.contextName1 }} clustermesh status --wait --wait-duration=5m
           cilium --context ${{ env.contextName2 }} clustermesh status --wait --wait-duration=5m
 
+      - name: Restart clustermesh-apiserver and ensure client can connect to new Service
+        run: |
+          echo "Restarting clustermesh-apiserver deployments"
+          kubectl --context ${{ env.contextName2 }} -n kube-system rollout restart deployment -l k8s-app=clustermesh-apiserver
+          kubectl --context ${{ env.contextName2 }} -n kube-system rollout status deployment -l k8s-app=clustermesh-apiserver
+          echo "Deploying a global Service to test failover"
+
+          kubectl --context ${{ env.contextName1 }} apply -f echo-failover.yaml
+          kubectl --context ${{ env.contextName2 }} apply -f echo-failover.yaml
+          echo "Testing client connection to global Service"
+
+          kubectl --context ${{ env.contextName1 }} -n cilium-test exec deploy/client -i -- curl -s -v --connect-timeout 5 --max-time 20 --retry-max-time 120 --retry-all-errors --retry 10 --fail echo-other-node-failover
```

Comment:
Timeouts should match the ones above.

Reply:
Sorry, EoD brain fry :sweat_smile: I was going to take a fresh look this morning before re-tagging your review